### ISSUE TITLE
Make cert check configurable per server

### ISSUE THREAD

**Issue by Epic428** [nzbgetcom/nzbget#53]
Since learning of nzbget development stopping, I switched to sabnzbd. Now that I've learned nzbget is picking back up again I figured I'd switch back over. One feature in sab that I like is the ability to configure certificate verification on a per server basis.

For whatever reason I just can't seem to get `super.newsgroupdirect.com` to work properly even when following the guide for handling self-signed certificates. (I get a hostname mismatch error, and when I ping the server it leads to `isp.news.ord.giganews.com`, however the cert is actually for `news.giganews.com`)

As a result, I have to resort to disabling certificate verification, which affects other servers where I don't have issues. It would be nice if I could disable the checking for one specific server rather than all of them.

**Comment by luckedea** [comment]
Hmm, this feature is implemented already under `feature-certificate-verification-level` but didn't make it into v22!
I couldn't find any news servers that needed it, so shelved it.

Great information, this gets into v23 for sure.

**Comment by dnzbk** [comment]
Hello @Epic428, thank you for your suggestion and for helping us make nzbget better.

I updated the _feature-certificate-verification-level_ branch and made test builds.
It would be great if you could help us to test this new feature and let us know if your problem has been solved.
To test this new feature you will need:

1. Delete an already installed version of the app.
2. Ddelete the old configuration files to avoid potential conflicts from one of these paths:
Windows: C:\ProgramData\NZBGet
macOS: <user>/Library/Application Support/NZBGet
Linux: the folder where the installation was performed.
3. Install the test version of the app:
macOS [build](https://github.com/nzbgetcom/nzbget/suites/18009665115/artifacts/1036283865)
Windows [build](https://github.com/nzbgetcom/nzbget/suites/18009665105/artifacts/1036286854)
Linux [build](https://github.com/nzbgetcom/nzbget/suites/18009665103/artifacts/1036349266)
4. Find a new option in: _Settings->NEWS-SERVERS->CertVerification_ and test it.

If everything goes well we will add this feature into v23.

Thank you in advance.

**Comment by Epic428** [comment]
Ah cool. I can test that later tonight and let you know how I make out. Thank you!

**Comment by Epic428** [comment]
I installed the test build on a ubuntu vm so I didn't have to worry about messing with my existing setup. When I try to test the connection the test fails for self-signed certificate. I can save the changes however and then drop an nzb file in and have it download successfully without any cert errors or warnings. I would maybe consider altering the text of the self-signed cert error or at least mention in the docs about the error displaying when being tested even if cert verification is turned off. Alternatively, returning successful/unsuccessful when the test button is clicked and the verification mode has been changed might be a better option overall. Otherwise from my quick test case I would say it works as expected.
![image](https://github.com/nzbgetcom/nzbget/assets/291578/090fa400-9a8e-46c4-93c5-19669f49f99e)
![image](https://github.com/nzbgetcom/nzbget/assets/291578/877542e5-34e3-43fe-8f74-a449ce58ec24)
